manifest/v1alpha/slo/validation.go

```diff
@@ -282,6 +282,23 @@ var anomalyConfigValidation = govy.New[AnomalyConfig](
 						WithName("project").
 						Rules(rules.StringDNSLabel()),
 				)),
+			govy.Transform(func(a AnomalyConfigNoData) string { return a.AlertAfter },
+				func(alertAfter string) (time.Duration, error) {
+					value, err := time.ParseDuration(alertAfter)
+					if err != nil {
+						return 0, err
+					}
+					if alertAfter != "" && value == 0 {
+						return 0, minimalNoDataAlertAfterRule.Validate(value)
+					}
+					return value, err
+				}).
+				WithName("alertAfter").
+				OmitEmpty().
+				Rules(
+					rules.DurationPrecision(time.Minute),
+					minimalNoDataAlertAfterRule,
+					rules.LTE(31*time.Hour*24)),
```

Comment:
While this is a valid working approach, I think we can improve it a bit :)
I think the main culprit here is that `AlertAfter` field should really be a pointer, If we make that change it will become easier to create a cleaner validation flow:

```suggestion
			govy.ForPointer(func(a AnomalyConfigNoData) *string { return a.AlertAfter }).
				WithName("alertAfter").
				Include(govy.New[string](
					govy.Transform(govy.GetSelf[string](), time.ParseDuration).
						Rules(
							rules.DurationPrecision(time.Minute),
							minimalNoDataAlertAfterRule,
							rules.LTE(31*time.Hour*24)),
				)),
```

Reply:
Oh nice, and now validation error is more meaningful:
```
ENCOUNTERED:
        	            	[
        	            	 {
        	            	  "propertyName": "spec.anomalyConfig.noData.alertAfter",
        	            	  "propertyValue": "0s",
        	            	  "errors": [
        	            	   {
        	            	    "error": "should be greater than or equal to '5m0s'",
        	            	    "code": "greater_than_or_equal_to",
        	            	    "description": "should be greater than or equal to '5m0s'"
        	            	   }
        	            	  ]
        	            	 }
        	            	]
```
compared to what it was before:
```
{
        	            	 "property": "spec.anomalyConfig.noData.alertAfter",
        	            	 "code": "transform",
        	            	 "message": "should be greater than or equal to '5m0s'"
        	            	}
```